src/Microsoft.AspNet.Razor.Test.Sources/CaseSensitiveTagHelperDescriptorComparer.cs

```diff
@@ -60,22 +60,25 @@ public override int GetHashCode(TagHelperDescriptor descriptor)
                     TagHelperDesignTimeDescriptorComparer.Default.GetHashCode(descriptor.DesignTimeDescriptor));
             }
 
-            foreach (var requiredAttribute in descriptor.RequiredAttributes)
+            foreach (var requiredAttribute in descriptor.RequiredAttributes.OrderBy(attribute => attribute))
             {
                 hashCodeCombiner.Add(requiredAttribute, StringComparer.Ordinal);
             }
 
             if (descriptor.AllowedChildren != null)
             {
-                foreach (var child in descriptor.AllowedChildren)
+                foreach (var child in descriptor.AllowedChildren.OrderBy(child => child))
                 {
                     hashCodeCombiner.Add(child, StringComparer.Ordinal);
                 }
             }
 
-            foreach (var attribute in descriptor.Attributes)
+            var orderedAttributeHashCodes = descriptor.Attributes
+                .Select(attribute => TagHelperAttributeDescriptorComparer.Default.GetHashCode(attribute))
+                .OrderBy(hashcode => hashcode);
```

Comment:
Little odd not to use the `TagHelperAttributeDescriptorComparer.Default` here (as in `Equals()`). But, sure, why not...